nucypher/blockchain/eth/sol/source/contracts/StakingEscrow.sol

```diff
@@ -583,7 +583,7 @@ contract StakingEscrow is Issuer {
         // initial stake of the staker
         if (info.subStakes.length == 0) {
             stakers.push(_staker);
-            policyManager.register(_staker, getCurrentPeriod());
+            policyManager.register(_staker, getCurrentPeriod() - 1);
```

Comment:
What's the implication of this change? Was it wrong before? This is setting `nodeInfo.lastMinedPeriod = currentPeriod - 1` in PolicyManager for each new staker, which at first sight looks weird, since technically, this staker didn't staked before.
Related to this, can you elaborate on what `nodeInfo.lastMinedPeriod` is and how is it used?

Reply:
`lastMinedPeriod` is field to store last period which was in `mint`, main purpose of this field is to loop through `rewardDelta[i]` only once for each period. First value for each node could be zero but in the result first `updateReward` will look into each `rewardDelta[i]` starting from 0, and a lot of periods will be even before PolicyManager creation. So we need to choose some period for `lastMinedPeriod` and the most important part - set period before any period when node can be used inside `createPolicy`. I choosed it wrong because thought only about active nodes (first active period is deposit period + 1, so it's why there was getCurrentPeriod()). And the result - `getCurrentPeriod() - 1` will solve it, because there is impossible to `createPolicy` for unexistent node.  Also I added some checks to make `PolicyManager` consistent even if `StakingEscrow` will send wrong values